### PR TITLE
Interaction States - NxTransferList - RSC-1485

### PR DESCRIPTION
**Pull request by entroform**
https://issues.sonatype.org/browse/RSC-1485

### CODE REVIEW

**Comment by avnvbhatta** [comment]
We also need to make changes to the active state. The new background color is grey-97 and border-color is grey-30

File: lib/src/components/NxTransferListHalf/NxTransferListHalf.scss [hunk 1]
```diff
@@ -115,12 +115,14 @@ $-visible-item-count: 10;
     padding-left: var(--nx-spacing-2x);
 
     &:hover {
+      background-color: var(--nx-color-interactive-background-hover);
```
Comment (EmilyBermudez88) [inline lib/src/components/NxTransferListHalf/NxTransferListHalf.scss:118]: It looks like on hover & click states, the text colour changes to nx-grey-50 as well.

File: lib/src/components/NxTransferListHalf/NxTransferListHalf.scss [hunk 2]
```diff
@@ -115,12 +115,14 @@ $-visible-item-count: 10;
...
-    &:focus-within {
-      border-color: var(--nx-color-interactive-border-focus);
-      box-shadow: var(--nx-box-shadow-focus);
+    &:focus-within:not(:active) {
+      outline: 2px solid var(--nx-color-interactive-border-focus);
+      outline-offset: -2px;
+      border-color: transparent;
     }
   
     &:active {
```
Comment (EmilyBermudez88) [inline lib/src/components/NxTransferListHalf/NxTransferListHalf.scss:128]: background colour here is nx-grey-97 instead of indigo

**Comment by EmilyBermudez88** [comment]
It doesn't stem from this ticket, but I'm also noticing that `nx-transfer-list__item` is 42px high in the gallery, but 40px high in the mockups. Perhaps we should match the heights in this ticket while we are changing the CSS :thinking_face:

File: gallery/visualtests/nxTransferListHalf.js [hunk 3]
```diff
@@ -64,8 +64,8 @@ describe('NxTransferListHalf', function() {
   });
 
   it('puts a dark border on hovered items', hoverTest(simpleListSelector, secondItemSelector));
-  it('puts a blue border and glow on focused items', focusTest(simpleListSelector, secondItemSelector));
-  it('puts a blue border and blue glow on focused+hovered items',
+  it('puts a blue border on focused items', focusTest(simpleListSelector, secondItemSelector));
+  it('puts a blue border on focused+hovered items',
```
Comment (EmilyBermudez88) [inline gallery/visualtests/nxTransferListHalf.js:68]:
```suggestion
  it('puts a blue border and dark background on focused+hovered items',
```

File: gallery/visualtests/nxTransferListHalf.js [hunk 4]
```diff
@@ -64,8 +64,8 @@ describe('NxTransferListHalf', function() {
...
       focusAndHoverTest(simpleListSelector, secondItemSelector));
   it('puts a grey background on clicked items', clickTest(simpleListSelector, secondItemSelector));
```
Comment (EmilyBermudez88) [inline gallery/visualtests/nxTransferListHalf.js:70]:
```suggestion
  it('puts a light grey background on clicked items', clickTest(simpleListSelector, secondItemSelector));
```
(if the other test name suggestions are used - to distinguish between active & hover backgrounds)

**Comment by entroform** [comment]
> We also need to make changes to the active state. The new background color is grey-97 and border-color is grey-30

Thanks! Fixed here: 70823b4c869928a19b192731ad59f9427937642c

**Comment by entroform** [comment]
> I think this was a recent sketch change, but the + / x icons for each transfer list item are grey-05 instead of being the same colour as the text.

Updated that here: 70823b4c869928a19b192731ad59f9427937642c

**Comment by entroform** [comment]
> It doesn't stem from this ticket, but I'm also noticing that `nx-transfer-list__item` is 42px high in the gallery, but 40px high in the mockups. Perhaps we should match the heights in this ticket while we are changing the CSS?

Hmm I checked and it looks like it's 40px for me? 🤔

**Comment by EmilyBermudez88** [comment]
>Hmm I checked and it looks like it's 40px for me? 🤔

weird! This is what I see
Screenshot 2023-03-14 at 12 01 34 PM

**Comment by entroform** [comment]
> > Hmm I checked and it looks like it's 40px for me? 🤔
> 
> weird! This is what I see Screenshot 2023-03-14 at 12 01 34 PM

Oh ops! so sorry I miss-read I thought you meant the mockup is 42 because I always thought the item was 40:
Fixed here 320b0017b3f69de1f5727e1440b938bda036e3b0

**Review by EmilyBermudez88 — APPROVED** [review]
LGTM!

**Pull request merged**